### PR TITLE
Infrastructure / Environment / OS

### PR DESCRIPTION
**Pull request by plaindocs**

### CODE REVIEW

File: user/reference/overview.md [hunk 1]
```diff
@@ -37,6 +47,28 @@ The following table summarizes the differences between the virtual environments:
...
+
+![Infrastructure shown in hostname](/images/ui/what-infrastructure.png "Infrastructure shown in hostname")
+
+if it contains:
+
+* `ec2`, as in the previous example → the build ran in a container-based environment on Amazon EC2.
+* `gce` → the build ran in a sudo-enabled environment on Google Cloud Engine.
+* `jupiter` → macOS
+
+### For a particular .travis.yml configuration
```
Comment (MariadeAnton) [inline user/reference/overview.md]: So in a very generic way we can say: 

- a language specified + no `os:` key  = Linux
also `os: linux`
  - `services: docker`,  `sudo: required`, `sudo: true` or any sudo commands in your .travis.yml file will send you to the fully virtualized Linux environment.

   - `sudo: false` or `no key specified` and no sudo commands in a `.travis.yml` - sends you to the container-based Linux environment

- `os: osx` or any of the `osx_image:` keys send you to the macOS infrastructure

File: user/reference/overview.md [hunk 2]
```diff
@@ -37,6 +47,41 @@ The following table summarizes the differences between the virtual environments:
...
+* `gce` → the build ran in a sudo-enabled environment on Google Cloud Engine.
+* `jupiter` → the build ran on macOS
+
+### For a particular .travis.yml configuration
+
+Many different parts of your `.travis.yml` affect what infrastructure your build runs on. The following list describes some of the main settings that determine build routing:
+
+* Any of the following "sudo related settings" route your build to a sudo-enabled linux (Ubuntu Trusty) environment on Google Cloud Engine.
+
+  - `services: docker`
```
[the remark this answers is not in the document]
Comment (plaindocs) [reply user/reference/overview.md]: But but but, that command contains the string docker but is not a docker command. 

Anyway, I've added #4674481d

File: user/reference/overview.md [hunk 3]
```diff
@@ -37,6 +47,42 @@ The following table summarizes the differences between the virtual environments:
...
+Many different parts of your `.travis.yml` affect what infrastructure your build runs on. The following list describes some of the main settings that determine build routing:
+
+* Any of the following sudo or docker related settings route your build to a sudo-enabled linux (Ubuntu Trusty) environment on Google Cloud Engine.
+
+  - `services: docker`
+  - `sudo: required` or `sudo: true`
+  - *any* other `sudo` command in your build script
+  _ *any* other `docker` command in your build script
+
+* Using `os: osx`, setting a version of OS X using `osx_image:`, or using a macOS specific language such as `language: objective-c` routes your build to macOS infrastructure.
```
Comment (cotsog) [inline user/reference/overview.md]: I know it's confusing but osx_image will rather set a version of Xcode.

Comment (plaindocs) [reply user/reference/overview.md]: Thanks @cotsog. Fixed.

**Pull request merged**